packages/core/ledger/ledger.go

```diff
@@ -192,7 +186,8 @@ func (l *Ledger) PruneTransaction(txID utxo.TransactionID, pruneFutureCone bool)
 	l.Storage.pruneTransaction(txID, pruneFutureCone)
 }
 
-func (l *Ledger) ForEachAcceptedUnSpentOutputWithMetadata(consumer func(*OutputWithMetadata)) {
+// ForEachAcceptedUnspentOutputWithMetadata returns the Accepted unspent OutputWithMetadata before a latest confirmed epoch.
```

Comment:
Is this what the method does? The Ledger should have no knowledge of the state of the epochs. Besides: the ledgerstate you should dump into the snapshot is the one contained in the notarization manager, not the one in the ledger.